File: app/src/main/java/com/example/leo/test/MainActivity.java
package com.example.leo.test;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;


public class MainActivity extends BaseActivity implements IOpenMediaHelper {
    private final String TAG =
            Utils.getTag(this);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i(TAG, "onCreate");

        setContentView(R.layout.activity_main);
    }

    @Override
    public void openMedia(String path) {
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragmentContainer, MediaFragment.newInstance(path))
                .addToBackStack(null)
                .commit();
    }

    @Override
    public String lastMedia() {
        return null;
    }

    @Override
    public String nextMedia() {
        return null;
    }
}

File: app/src/main/kotlin/com/example/leo/test/MainActivity.kt
package com.example.leo.test

import android.os.Bundle
import android.util.Log

interface IOpenMediaHelper {
    fun openMedia(path: String)
    fun lastMedia(): String?
    fun nextMedia(): String?
}

class MainActivity : BaseActivity(), IOpenMediaHelper {
    private val TAG = Utils.getTag(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i(TAG, "onCreate")

        setContentView(R.layout.activity_main)
    }

    override fun openMedia(path: String) {
        supportFragmentManager.beginTransaction()
                .replace(R.id.fragmentContainer, MediaFragment.newInstance(path))
                .addToBackStack(null)
                .commit()
    }

    override fun lastMedia(): String? {
        return null
    }

    override fun nextMedia(): String? {
        return null
    }
}
